### ISSUE TITLE
[suggestion] Separate binary dependencies 

### ISSUE THREAD

**Issue by makapuf** [axiros/rusp#11]
There are several ways to to simplify build and not require many dependencies not needed for library usage, see by example: 

https://stackoverflow.com/questions/35711044/how-can-i-specify-binary-only-dependencies

**Comment by therealprof** [comment]
While that is true, you might have noticed that the few dependencies we have (serde, quick-protobuf, clap and anyhow) are used even in the library part. It might be possible to make that leaner but that doesn't seem to warrant the effort. At least our library users also have the same dependencies themselves.

**Comment by makapuf** [comment]
You're right of course : I wasn't using those parts, focusing on serializing / deserializing USP without encoding it to json or using the commandline.

**Comment by thalesfragoso** [comment]
I believe this is now done. We have a separated lib with minimal dependencies.
